pkg/frontend/providers_validation.go

```diff
@@ -0,0 +1,60 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+
+	mgmtfeatures "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-07-01/features"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/features"
+)
+
+type ProvidersValidator interface {
+	ValidateProviders(ctx context.Context, azEnv *azureclient.AROEnvironment, environment env.Interface, subscriptionID, tenantID string) error
+}
+
+type providersValidator struct{}
+
+func (p providersValidator) ValidateProviders(ctx context.Context, azEnv *azureclient.AROEnvironment, environment env.Interface, subscriptionID, tenantID string) error {
+	fpAuthorizer, err := environment.FPAuthorizer(tenantID, environment.Environment().ResourceManagerEndpoint)
+	if err != nil {
+		return err
+	}
+
+	providersClient := features.NewProvidersClient(azEnv, subscriptionID, fpAuthorizer)
+
+	return validateProviders(ctx, providersClient)
+}
+
+func validateProviders(ctx context.Context, providersClient features.ProvidersClient) error {
```

Comment:
This duplicate name might be confusing. I am tempted to say that we don't need this extra function since we are just 1) authorizing and then 2) checking against a list, we could do it all in ValidateProviders() above

Reply:
to provide some context, this is how I did it in quotavalidation and this is (I think) why he did it this way. The reason why I did it this way is that it's not necessarily meaningful to test it with all of these parameters, and makes it harder to set up in the tests (in the tests we only test the private method). Now, if you want to change it, I don't really mind if you found a better way (but then we might want to apply the change to quotavalidator and invalidparameters)

Reply:
I did copy your pattern for QuotaValidation basically wholesale for this PR, so that is why I have the same setup. It also allowed me to migrate the validation unit tests as-is, since the private function shared the same signature as the old validation function.  

I agree that long-term we should inline these private functions, but should find a pattern to do it for all of the new frontend validators.